controllers/UserController.ts

```diff
@@ -107,9 +109,35 @@ export default class UserController implements UserControllerI {
      * @param {Response} res Represents response to client, including status
      * on whether updating a user was successful or not
      */
-    updateUser = (req: Request, res: Response) =>
-        UserController.userDao.updateUser(req.params.uid, req.body)
-            .then(status => res.json(status))
+    updateUser = async (req: Request, res: Response) => {
+        const userInfo = req.body
+        let existingUser = false
+
+        // checks if username is taken
+        if (userInfo.username) {
+            existingUser = await UserController.userDao.findUserByUsername(userInfo.username)
+        }
+
+        // if no username taken or changed
+        if (!existingUser) {
+            // encrypt password if changed
+            if (userInfo.password) {
+                const password = userInfo.password;
+                userInfo.password = await bcrypt.hash(password, saltRounds);
+            }
+
+            // update the user
+            await UserController.userDao.updateUser(req.params.uid, userInfo).then(status => res.json(status))
```

Comment:
I will pass the profile with "_id", perhaps we don't need uid to be in the api url for security.

Reply:
keeping as is for now